components/user/EventRow.tsx

```diff
@@ -0,0 +1,297 @@
+          ? styles.active
+          : ''
+      )}
+    >
+      <>
+        Block &hellip; {abbrevHash}
+        <ActivityCopy className="ml-4" />
+      </>
+    </div>
+  )
+}
+
+const summarizeEvent = (
+  type: EventType,
+  metadata: ApiEventMetadata
+): ReactElement => {
+  if (type === EventType.BLOCK_MINED) {
+    // return 'View in the explorer'
+    const { hash } = metadata as ApiEventMetadataBlockMined
+    return <CopyableHash hash={hash} />
+  }
+  const { url } = metadata as ApiEventMetadataWithLink
+  const link = new URL(url)
+  const { pathname, hostname } = link
+  const parts = pathname.split('/')
+  const id = parts.slice(-1)[0]
+  if (type === EventType.PULL_REQUEST_MERGED) {
+    // https://github.com/iron-fish/ironfish-api/pull/595
+    return <>View pull request #{id}</>
+  } else if (type === EventType.BUG_CAUGHT) {
+    // https://github.com/iron-fish/ironfish/issues/930
+    return <>View issue #{id}</>
+  } else if (type === EventType.COMMUNITY_CONTRIBUTION) {
+    return <>View contribution</>
+  } else if (type === EventType.SOCIAL_MEDIA_PROMOTION) {
+    return <>Promoted on {hostname}</>
+  }
+  return <>UNHANDLED: {type}</>
+}
+
+const formatEventDate = (d: Date) =>
+  format(d, `y'-'MM'-'dd HH':'mm':'ss`, {
+    locale: enUS,
+  })
+
+export const EventRow = ({
+  type,
+  occurred_at: occurredAt,
+  points,
+  metadata,
+}: EventRowProps) => {
+  const { text, icon } = displayEventType(type)
+
+  const eType = (
+    <div className="flex items-center justify-start">
+      <span className="mr-2">{icon}</span>
+      {text}
+    </div>
+  )
+
+  return (
+    <tr className="border-b border-black">
+      <td className="py-4 w-1/3">{eType}</td>
+      <td>{formatEventDate(new Date(occurredAt))}</td>
+      <td>{points}</td>
+      <td className="max-w-[11rem]">
+        <a
+          href={makeLinkForEvent(type, metadata)}
+          className="text-ifotherblue align-left flex items-end justify-between"
+          target="_blank"
+          rel="noreferrer"
+        >
+          <div>{metadata && summarizeEvent(type, metadata)}</div>
+          <ChevronRight />
+        </a>
+      </td>
+    </tr>
+  )
+}
+
+type WeekRowProps = {
+  date: Date
+  week: number
+}
+
+const WeekRow = ({ date, week }: WeekRowProps) => {
+  const when = `Week ${week} - Started ${format(date, 'MMMM do, Y')}`
+  return (
+    <tr
+      className="bg-black text-white"
+      data-date={date}
+      aria-label={when}
+      title={when}
+    >
+      <td
+        colSpan={4}
+        className="text-center uppercase text-xs tracking-widest h-8"
+      >
+        Week {week}
+      </td>
+    </tr>
+  )
+}
+const weeksBetween = (start: Date, end: Date) =>
+  eachWeekOfInterval({ start, end })
+
+const eventsBetween = (
+  start: Date,
+  end: Date,
+  events: ApiEvent[]
+): ApiEvent[] =>
+  events.filter(e => {
+    const time = parseISO(e.occurred_at)
+    return isAfter(time, start) && isBefore(time, end)
```

Comment:
I'm not 100% sure that the time bounding is right here -- Looking at this user: https://website-testnet-4mat9grpo-ironfish.vercel.app/users/1  The first event in Week 3 is `2021-12-05 01:00:14` (which I assume is PST), which as UTC is `Sun, 05 Dec 2021 09:00:14 GMT`.

Based on the leaderboard, the weeks are supposed to roll over on Monday at midnight, UTC.

Reply:
I adjusted things after finding a `weekStartsOn` parameter: https://github.com/iron-fish/website-testnet/pull/168/files#diff-e2896aa815d18da702e45b20bd81e18ddf7bf6c79623a7b67a21a0828833da31R237

![CleanShot 2022-02-07 at 13 12 37](https://user-images.githubusercontent.com/18919/152872997-9d6ceefb-6367-4a03-8526-aff1943c49eb.png)

The only potential outlier is that the week of our first week started on Monday the 29th of December, but I think it's now working correctly.

Reply:
Nice, it's close! The week range still needs to use UTC -- you can check it by looking on https://website-testnet-git-feature-iro-1420-week-of-activity-ironfish.vercel.app/users/1 -- at the end of some weeks, like Week 1 and Week 3, points start getting counted again, and if you take those times and convert them to UTC, they'll be early in the morning on Monday UTC.